js/common/lib/backend-impl.ts

```diff
@@ -109,17 +112,28 @@ const tryResolveAndInitializeBackend = async(backendName: string): Promise<Backe
  */
 export const resolveBackendAndExecutionProviders = async(options: InferenceSession.SessionOptions):
     Promise<[backend: Backend, options: InferenceSession.SessionOptions]> => {
-      // extract backend hints from session options
+      // extract backend hints from session options.
       const eps = options.executionProviders || [];
       const backendHints = eps.map(i => typeof i === 'string' ? i : i.name);
       const backendNames = backendHints.length === 0 ? backendsSortedByPriority : backendHints;
-
-      // try to resolve and initialize all requested backends
+      if (backendNames.filter(name => name === 'webgpu').length > 1) {
+        throw new Error(`Registering duplicate 'webgpu' backends in the session options is not permitted`);
+      }
+      if (backendNames.filter(name => name === 'webnn').length > 1) {
+        throw new Error(`Registering duplicate 'webnn' backends in the session options is not permitted`);
+      }
```

Comment:
Will this disable trying do this?

```js
  executionProviders: [
    {
      name: "webnn",
      deviceType: "gpu",
    },
    {
      name: "webnn",
      deviceType: "cpu",
    },
    ...
  ]
```

Reply:
Creating duplicated WebNN or WebGPU eps in a single session is not allowed, @fs-eire, right?

Reply:
I don't know if that is possible or not technically. But in practice, I think there is no use case of multiple instances of the same EP being used in one session.